File: backend/controllers/VendorController.js
const VendorService = require("../services/VendorService");

// Add Vendor
const addVendor = async (req, res, next) => {
    try {
        const result = await VendorService.addVendor(req.body);

        res.status(201).json({
            success: true,
            message: "Vendor added successfully",
            data: result,
        });
    } catch (error) {
        next(error);
    }
};

// Get All Vendors
const getVendors = async (req, res, next) => {
    try {
        const vendors = await VendorService.getVendors();

        res.status(200).json({
            success: true,
            data: vendors,
        });
    } catch (error) {
        next(error);
    }
};

// Update Vendor
const updateVendor = async (req, res, next) => {
    try {
        const { id } = req.params;

        await VendorService.updateVendor(id, req.body);

        res.status(200).json({
            success: true,
            message: "Vendor updated successfully",
        });
    } catch (error) {
        next(error);
    }
};

// Delete Vendor
const deleteVendor = async (req, res, next) => {
    try {
        const { id } = req.params;

        await VendorService.deleteVendor(id);

        res.status(200).json({
            success: true,
            message: "Vendor deleted successfully",
        });
    } catch (error) {
        next(error);
    }
};

module.exports = {
    addVendor,
    getVendors,
    updateVendor,
    deleteVendor,
};

File: backend (1)/backend/src/main/java/com/procureease/backend/controller/VendorController.java
package com.procureease.backend.controller;

import com.procureease.backend.dto.VendorRequest;
import com.procureease.backend.dto.VendorResponse;
import com.procureease.backend.service.VendorService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/vendors")
@RequiredArgsConstructor
public class VendorController {

    private final VendorService vendorService;

    // ============================
    // Create Vendor
    // ============================

    @PostMapping
    public ResponseEntity<Map<String, Object>> createVendor(
            @Valid @RequestBody VendorRequest request
    ) {

        VendorResponse vendor = vendorService.createVendor(request);

        Map<String, Object> response = new HashMap<>();

        response.put("success", true);
        response.put("message", "Vendor added successfully");
        response.put("data", vendor);

        return ResponseEntity
                .status(HttpStatus.CREATED)
                .body(response);
    }

    // ============================
    // Get All Vendors
    // ============================

    @GetMapping
    public ResponseEntity<Map<String, Object>> getAllVendors() {

        List<VendorResponse> vendors =
                vendorService.getAllVendors();

        Map<String, Object> response = new HashMap<>();

        response.put("success", true);
        response.put("data", vendors);

        return ResponseEntity.ok(response);
    }

    // ============================
    // Update Vendor
    // ============================

    @PutMapping("/{id}")
    public ResponseEntity<Map<String, Object>> updateVendor(
            @PathVariable Integer id,
            @Valid @RequestBody VendorRequest request
    ) {

        vendorService.updateVendor(id, request);

        Map<String, Object> response = new HashMap<>();

        response.put("success", true);
        response.put("message", "Vendor updated successfully");

        return ResponseEntity.ok(response);
    }

    // ============================
    // Delete Vendor
    // ============================

    @DeleteMapping("/{id}")
    public ResponseEntity<Map<String, Object>> deleteVendor(
            @PathVariable Integer id
    ) {

        vendorService.deleteVendor(id);

        Map<String, Object> response = new HashMap<>();

        response.put("success", true);
        response.put("message", "Vendor deleted successfully");

        return ResponseEntity.ok(response);
    }
}